File: src/JavaScript/fibonacci.js
const fibonacciRecursive =(n) =>{
    if (n <= 0) return 0;
    if (n == 1) return 1;
    return fibonacciRecursive(n - 1) + fibonacciRecursive(n - 2);
};
const fibonacciIterative =(n) =>{
    if (n <= 0) return 0;
    if (n == 1) return 1;

    var a = 0, b = 1;
    for (var i = 2; i <= n; i++) {
        var temp = a + b;
        a = b;
        b = temp;
    }
    return b;
};
const n = 10;
var answer1 = fibonacciRecursive(n);
var answer2 = fibonacciIterative(n);

console.log("Fibonacci recursif ("+n+") : " + answer1);
console.log("Fibonacci iteratif ("+n+") :" + answer2);

File: src/C++/fibonacci.cpp
#include <iostream>
#include <cstdint>

// Function prototypes
int fibonacciRecursive(int n);
int fibonacciIterative(int n);

int main() {
    int n = 10;
    int answer1 = fibonacciRecursive(n);
    int answer2 = fibonacciIterative(n);

    std::cout << "Fibonacci recursif (" << n << ") : " << answer1 << std::endl;
    std::cout << "Fibonacci iteratif (" << n << ") : " << answer2 << std::endl;

    return 0;
}

// Recursive Fibonacci function
int fibonacciRecursive(int n) {
    if (n <= 0) return 0;
    if (n == 1) return 1;
    return fibonacciRecursive(n - 1) + fibonacciRecursive(n - 2);
}

// Iterative Fibonacci function
int fibonacciIterative(int n) {
    if (n <= 0) return 0;
    if (n == 1) return 1;

    int a = 0, b = 1;
    for (int i = 2; i <= n; i++) {
        int temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}